domains/transfer/Transfer.API/API/Transfer/Api/Controllers/TransferAgreementProposalController.cs

```diff
@@ -0,0 +1,138 @@
+            EndDate = request.EndDate == null ? null : DateTimeOffset.FromUnixTimeSeconds(request.EndDate.Value)
+        };
+
+        var hasConflict = await transferAgreementRepository.HasDateOverlap(newProposal);
+
+        if (hasConflict)
+        {
+            return Conflict("There is already a Transfer Agreement with this company tin within the selected date range");
+        }
+
+        await repository.AddTransferAgreementProposal(newProposal);
+
+        return CreatedAtAction(nameof(GetTransferAgreementProposal), new { id = newProposal.Id }, newProposal);
+    }
+
+    /// <summary>
+    /// Get TransferAgreementProposal by Id
+    /// </summary>
+    /// <param name="id">Id of TransferAgreementProposal</param>
+    /// <response code="200">Successful operation</response>
+    /// <response code="400">You cannot Accept/Deny your own TransferAgreementProposal, you cannot Accept/Deny a TransferAgreementProposal for another company or this proposal has run out</response>
+    /// <response code="404">TransferAgreementProposal expired or deleted</response>
+    [ProducesResponseType(typeof(TransferAgreementProposal), 200)]
+    [ProducesResponseType(typeof(void), 400)]
+    [ProducesResponseType(typeof(void), 404)]
+    [HttpGet("{id}")]
+    public async Task<ActionResult<TransferAgreementProposal>> GetTransferAgreementProposal(Guid id)
+    {
+        var proposal = await repository.GetNonExpiredTransferAgreementProposal(id);
+
+        if (proposal == null)
+        {
+            return NotFound("TransferAgreementProposal expired or deleted");
+        }
+
+        var currentCompanyId = Guid.Parse(User.FindSubjectGuidClaim());
+        var currentCompanyTin = User.FindSubjectTinClaim();
+
+        if (currentCompanyId == proposal.SenderCompanyId)
+        {
+            return BadRequest("You cannot Accept/Deny your own TransferAgreementProposal");
+        }
+        if (currentCompanyTin != proposal.ReceiverCompanyTin)
+        {
+            return BadRequest("You cannot Accept/Deny a TransferAgreementProposal for another company");
+        }
+        if (proposal.EndDate < DateTimeOffset.UtcNow)
+        {
+            return BadRequest("This proposal has run out");
+        }
+
+        return Ok(proposal);
+    }
+
+    /// <summary>
+    /// Delete TransferAgreementProposal
+    /// </summary>
+    /// <param name="id">Id of TransferAgreementProposal</param>
+    /// <response code="204">Successful operation</response>
+    /// <response code="404">TransferAgreementProposal not found</response>
+    [ProducesResponseType(typeof(void), 204)]
+    [ProducesResponseType(typeof(void), 404)]
+    [HttpDelete("{id}")]
+    public async Task<ActionResult> DeleteTransferAgreementProposal(Guid id)
+    {
+        var proposal = await repository.GetNonExpiredTransferAgreementProposal(id);
+
+        if (proposal == null)
+        {
+            return NotFound("TransferAgreementProposal not found");
```

Comment:
According to our guidelines 404 must not contain any body

Reply:
corrected